tls/_constructs.py

```diff
@@ -137,21 +144,24 @@
 
 ASN1Cert = Struct(
     "ASN1Cert",
-    UBInt32("length"),   # TODO: Reject packets with length not in 1..2^24-1
+    SizeWithin(UBInt32("length"),
+               min_size=1, max_size=2 ** 24 - 1),
     Bytes("asn1_cert", lambda ctx: ctx.length),
 )
 
 Certificate = Struct(
-    "Certificate",  # TODO: Reject packets with length > 2 ** 24 - 1
-    UBInt32("certificates_length"),
+    "Certificate",
+    SizeWithin(UBInt32("certificates_length"),
+               min_size=1, max_size=2 ** 24 - 1),
     Bytes("certificates_bytes", lambda ctx: ctx.certificates_length),
 )
 
 Handshake = Struct(
     "Handshake",
     UBInt8("msg_type"),
-    UBInt24("length"),  # TODO: Reject packets with length > 2 ** 24
```

Comment:
`UBInt24`s can't hold a value greater than 2 ** 24 - 1.  If given something longer there could be some misread data, but I don't know how to detect this (e.g., does `\xff\xff\xff\xff` mean "this is an array of length of 2 ** 24 - 1, and its first element is 255" or "this is an array of length 2 ** 32 - 1"?  Is there a way to tell??)